templates/demo.html

```diff
@@ -7,46 +7,75 @@
 {% block content %}
 <script type="module">
   import AnboxStream from '/static/js/anbox-stream-sdk.js';
+  // @ params appName: name of the application to stream
+  function streamContainer(appName, height = false, width = false) {
+    var container = document.getElementById(appName);
+    var errorContainer = container.getElementsByClassName("js-error-container")[0];
+    var errorMessage = container.getElementsByClassName("js-error")[0];
+    var button = container.getElementsByClassName("js-button")[0];
+    var player = document.getElementById(appName + "-player");
+    var loading = container.getElementsByClassName("js-loading")[0];
+    var play = container.getElementsByClassName("js-play")[0];
 
-  let stream = new AnboxStream({
-  targetElement: "anbox-stream",
-  url: '{{ ANBOXCLOUD_API_BASE }}',
-  authToken: '{{ session["authentication_token"] }}',
-  session: {
-      app: "com.vectorunit.cobalt.canonical", // Given
-  },
-  screen: {
-    // Currently available max resolution
-    // To change to portrait just flip the values between width and height
-      width: 1280,
-      height: 720,
-  },
-  // Using keyboard controls to test it on non-touch device
-  controls: {
-    keyboard: true
-  },
-  callbacks: {
-    error: error => {
-      console.log("AnboxStream failed: " + error.message);
-    }
+    play.classList.add("u-hide");
+    loading.classList.remove("u-hide");
+
+    let stream = new AnboxStream({
+      targetElement: player.id,
+      url: '{{ ANBOXCLOUD_API_BASE }}',
+      authToken: '{{ session["authentication_token"] }}',
+      session: {
+        app: "com.vectorunit.cobalt.canonical", // Given
+      },
+      screen: {
+        width: width && width < 1104 ? width : 1104,
+        height: height && height < 628 ? height : 628,
+        fps: 25
+      },
+      // Using keyboard controls to test it on non-touch device
+      controls: {
+        keyboard: true
+      },
+      callbacks: {
+        error: error => {
+          button.classList.remove("u-hide");
+          player.classList.add('u-hide');
+          loading.classList.add("u-hide");
+          play.classList.remove("u-hide");
+          errorMessage.textContent = error + ". Please try again.";
```

Comment:
Errors can be pretty user-unfriendly, could we just print a generic error and log the error ?

Reply:
Yeah, we can do that. Or if there are specific ones we'd like to catch and display, that's also an option.

Reply:
I believe we can group the errors and display more user-friendly generic errors. But I think to do this, will we need to know which error is which independently of the text?

Reply:
I don't think there's any error that could require a user action in the SDK. It could be interesting to gather logs from the SDK to know what went wrong but that's a completely different story

Reply:
We have sometimes `Connection closed` errors which can be fixed by simply refreshing the browser window. Sometimes even the `Failed to join` also can be solved by refreshing, but as Simon said before, avoiding triggering multiple anbox streams could fix this, so shouldn't the user be aware of this?

Reply:
The issue Simon mentionned was when a user tried to refresh right after stopping a stream. We fixed that issue by returning an error if a user tries to restart a stream too soon. We could have a generic error message that basically says `please refresh` or `please try again`, but displaying the error content is not appropriate here

Reply:
Ok that makes sense. Thanks